File: nagasenacs/src/Nagasena/Proc/IO/ValueScannerBase.cs
using System.IO;

namespace Nagasena.Proc.IO {

  using QName = Nagasena.Proc.Common.QName;
  using EXISchema = Nagasena.Schema.EXISchema;

  internal abstract class ValueScannerBase : ValueScanner {

    private readonly QName m_name;

    internal ValueScannerBase(QName name) {
      m_name = name;
    }

    public override int getBuiltinRCS(int simpleType) {
      return EXISchema.NIL_NODE;
    }

    public override sealed Stream InputStream {
      set {
        m_istream = value;
      }
    }

    public override QName Name {
      get {
        return m_name;
      }
    }

  }

}

File: nagasena/src/org/openexi/proc/io/ValueScannerBase.java
package org.openexi.proc.io;

import java.io.InputStream;

import org.openexi.proc.common.QName;
import org.openexi.schema.EXISchema;

abstract class ValueScannerBase extends ValueScanner {
  
  private final QName m_name;

  ValueScannerBase(QName name) {
    m_name = name;
  }
 
  @Override
  public int getBuiltinRCS(int simpleType) {
    return EXISchema.NIL_NODE;
  }

  @Override
  public final void setInputStream(InputStream istream) {
    m_istream = istream;
  }

  @Override
  public final QName getName() {
    return m_name;
  }

}
